cmake/Modules/FindBoost.cmake

```diff
@@ -0,0 +1,61 @@
+# use static Boost in Windows
+if(WIN32)
```

Comment:
Why only WIN32? Curious

Reply:
No idea. Honestly, there is a lot of stuff that I don't understand why it exists 😅 , but better not touch it, might break some stuff.

There was another one @SergioRAgostinho has mentioned above: https://github.com/PointCloudLibrary/pcl/blob/17cc3de419edbee2368dc3d4ac18391ce67c8c8e/cmake/Modules/FindBoost.cmake#L28